### DISCUSSION TITLE
Enhancement issues

### DISCUSSION THREAD

**Discussion by gokadzev** [gokadzev/Musify#166] [Announcements]
All current issues labeled enhancements are moved into discussion sections.

**Comment by Zaxx69** [comment]
Hey brother can you make two things possible for musify
1. Let us play the downloaded songs directly from the song interface rather than going to local music. I mean can you make it online and offline. Pretty please brother
2. Let's us create our own playlist.
The app is great and we love the app but these features would be appreciated brother

**Reply by gokadzev** [reply]
Hello, sorry it seems like I missed your comment somehow. 

About the first one, I'm not sure what you mean. The app supports both online and offline playing so they should be divided somehow.

About the second one, It's already requested but kind of complex and when we have alternatives such as youtube playlist adding and liked songs I prefer not adding this.

**Reply by msooon** [reply]
For me it was not possible to play a downloaded song. It always tried to stream. But it would be great to play downloaded songs inside the app

**Reply by gokadzev** [reply]
@msooon Local music playing support was removed in V6.4.0. Providing both offline and online music playing support with all necessary features wasn't that easy, especially considering some dependency bugs that still exist and haven't been fixed yet. Although I am still considering adding the "make available for offline" feature.

**Reply by foozaweb** [reply]
I have a solution for downloading and playing songs offline. The songs could be downloaded to the `getApplicationDocumentsDirectory()` Directory making it only accessible via the app. I've tried to access it, but I have a problem with AudioHandler not being initialized when I try to play local songs

**Reply by gokadzev** [reply]
> I have a solution for downloading and playing songs offline. The songs could be downloaded to the `getApplicationDocumentsDirectory()` Directory making it only accessible via the app. I've tried to access it, but I have a problem with AudioHandler not being initialized when I try to play local songs

Actually, not a bad solution. The only thing I don't like is having the download and offline features together. If I'm able to implement it, it would be more useful, as the download implementation is currently a "better than nothing" solution.

**Reply by foozaweb** [reply]
Okay, I will work on that and update you then, but do you have a solution to the issue with `AudioHandler` not being initialized when I try to play local songs?

**Reply by gokadzev** [reply]
Depends on what you changed, it shouldn't have any connection with AudioHandler unless you changed it's implementation (which I think  we won't need to implement offline audio feature)

**Reply by foozaweb** [reply]
Ok. I think I understand what you mean now.
